src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/JsonConverters/ImageJsonConverter.cs

```diff
@@ -16,18 +16,13 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
             else
             {
                 var val = (Image)value;
-                writer.WriteStartObject();
-                writer.WritePropertyName("sourceImage");
-                writer.WriteValue(val.sprite?.name);
-                writer.WritePropertyName("color");
-                serializer.Serialize(writer, val.color, typeof(Color));
-                writer.WritePropertyName("material");
-                writer.WriteValue(val.material?.name);
-                writer.WritePropertyName("raycastTarget");
-                writer.WriteValue(val.raycastTarget);
-                writer.WritePropertyName("preserveAspect");
-                writer.WriteValue(val.preserveAspect);
-                writer.WriteEndObject();
+                // raw is way faster than using the libraries
+                writer.WriteRawValue("{\"sourceImage\":" + JsonConvert.ToString(val.sprite?.name)
```

Comment:
Just out of curiosity, does C# also have a StringBuilder concept like in Java? I wonder if these string constructions are creating lots of intermediate strings.

Reply:
they do, but + is faster for anything less than about 1000 appends in a code block given how the compiler optimizes + to be a concat operation

Reply:
if you want a detailed rundown on the performance details i'd be happy to share, but suffice to say, nothing so easy/shallow as this is going to solve the issue / I've 99.99% sure already considered and/or tested it